Browser_IDE/executionEnvironment_Internal.js

```diff
@@ -515,4 +515,38 @@ moduleEvents.addEventListener("onRuntimeInitialized", function() {
     });
 
     parent.postMessage({type:"initialized"},"*");
-});
\ No newline at end of file
+});
+
+function scaleCanvas() {
+    const container = document.getElementById('canvasContainer');
+    const canvas = document.getElementById('canvas');
+    
+    const maxWidth = container.clientWidth;
+    const maxHeight = container.clientHeight;
+    
+    const canvasWidth = canvas.width;
+    const canvasHeight = canvas.height;
+    
+    let scale = Math.min(maxWidth / canvasWidth, maxHeight / canvasHeight);
+    
+    canvas.style.width = `${canvasWidth * scale}px`;
+    canvas.style.height = `${canvasHeight * scale}px`;
+    
+    canvas.style.marginTop = `${(maxHeight - canvasHeight * scale) / 2}px`;
+    canvas.style.marginLeft = `${(maxWidth - canvasWidth * scale) / 2}px`;
+}
```

Comment:
`scaleCanvas` is not necessary. It can be removed. The width and height are already set properly by the runtime, and I believe these margin values will be equivalent in effect to `margin: auto`, as they will simply center the element within its container. The only reason that they would be necessary is because you have set the property `position: absolute` on the canvas element.